packages/teleterm/src/ui/TabHost/TabHost.tsx

```diff
@@ -28,22 +28,19 @@ import { ClusterConnectPanel } from './ClusterConnectPanel/ClusterConnectPanel';
 export function TabHostContainer() {
   const ctx = useAppContext();
   ctx.workspacesService.useState();
-
   const isRootClusterSelected = !!ctx.workspacesService.getRootClusterUri();
 
-  return useMemo(() => {
-    if (isRootClusterSelected) {
-      return <TabHost />;
-    }
-    return <ClusterConnectPanel />;
-  }, [isRootClusterSelected]);
+  if (isRootClusterSelected) {
+    return <TabHost />;
+  }
+  return <ClusterConnectPanel />;
 }
 
 export function TabHost() {
   const ctx = useAppContext();
   const documentsService =
     ctx.workspacesService.getActiveWorkspaceDocumentService();
-  const activeDocument = documentsService.getActive();
+  const activeDocument = documentsService?.getActive();
```

Comment:
Sometimes (very rarely for me) there was the error `cannot read property getActive of undefined` after logout. Theoretically, it should not be possible, because condition in `TabHostContainer` protects us from it - we can access `activeDocumentService` only when `rootClusterUri` is not null. However, the error was occurring. The reason for this is the order in which the hooks are called. React runs `useEffect` starting from children. It means that `TabHost` added itself to the `workspaceService` subscribers before `TabHostContainer`, so when `rootClusterUri` became `null` child component was notified before the parent and tried to re-render. 
I got rid of unnecessary subscriptions. I don't know if we should take to do something with it globally (like calling callbacks in reverse order, but that could potentially cause problems elsewhere).